app/src/main/java/com/android/voyageur/ui/trip/activities/ActivitiesMapScreen.kt

```diff
@@ -0,0 +1,275 @@
+ *
+ * @param onFilterByDay Callback for filtering activities by date.
+ * @param onFilterByType Callback for filtering activities by type.
+ * @param onClearFilters Callback for clearing all filters.
+ * @param onDismiss Callback for dismissing the dialog.
+ */
+@Composable
+fun FilterDialog(
+    onFilterByDay: (Date) -> Unit,
+    onFilterByType: (Set<String>) -> Unit,
+    onClearFilters: () -> Unit,
+    onDismiss: () -> Unit,
+    selectedFilters: Set<String>
+) {
+  val activityTypes = ActivityType.entries.map { it.toString() }
+  val context = LocalContext.current
+  var selectedDate by remember { mutableStateOf<Date?>(null) }
+  var tempSelectedFilters by remember { mutableStateOf(selectedFilters) }
+  var showTypeOptions by remember { mutableStateOf(false) }
+
+  if (showTypeOptions) {
+    // Secondary dialog for activity type options
+    AlertDialog(
+        modifier = Modifier.testTag("typeFilterActivityAlertDialog"),
+        onDismissRequest = { showTypeOptions = false },
+        title = { Text(stringResource(R.string.select_activities_by_type)) },
+        text = {
+          Column {
+            activityTypes.forEach { type ->
+              Row(
+                  modifier = Modifier.fillMaxWidth(),
+                  verticalAlignment = Alignment.CenterVertically) {
+                    val isChecked = tempSelectedFilters.contains(type)
+                    Checkbox(
+                        modifier = Modifier.testTag("checkbox_$type"),
+                        checked = isChecked,
+                        onCheckedChange = { isSelected ->
+                          tempSelectedFilters =
+                              if (isSelected) {
+                                tempSelectedFilters + type // Add type to set
+                              } else {
+                                tempSelectedFilters - type // Remove type from set
+                              }
+                        })
+                    Text(text = type, modifier = Modifier.padding(start = 8.dp))
+                  }
+            }
+          }
+        },
+        confirmButton = {
+          TextButton(
+              modifier = Modifier.testTag("applyTypeFilterButton"),
+              onClick = {
+                onFilterByType(tempSelectedFilters)
+                showTypeOptions = false
+              }) {
+                Text(stringResource(R.string.apply))
+              }
+        },
+        dismissButton = {
+          TextButton(onClick = { showTypeOptions = false }) {
+            Text(stringResource(R.string.cancel))
+          }
+        })
+  } else {
+    // Main dialog
+    AlertDialog(
+        onDismissRequest = onDismiss,
+        title = { Text(stringResource(R.string.filter_activities_button)) },
+        text = {
+          Column {
+            // Filter by Date Button
+            Button(
+                onClick = {
+                  val calendar = Calendar.getInstance()
+                  DatePickerDialog(
```

Comment:
Also when the date picker dialog is open, the primary dialog should be closed, as you can see that they overlap. 
Moreover, there is already a DatePicker component in the project in ui/formFields. Let me know if you think you can integrate or if it wouldn't work.

Reply:
![image](https://github.com/user-attachments/assets/fef5d1d7-d2f6-4c4e-b5e0-6a3786a2d1e6)

Reply:
It wouldn't work as our custom date picker is a composable so I cannot call it in the OnClick function of a button.

Reply:
And also about the main dialog being open when the date picker is chosen, the overlapping is not deranging and anyways per your previous comment it should be open after picking the date. Closing the main dialog when choosing the date then reopening it when the date picker is closing will do a weird animation (I tried it on the emulator) that I believe is more troubling then the minimal overlapping. But if you feel like it is a must then I will implement it.

Reply:
same thing, we can address it in the UI PR